code/code/misc/discord.cc

```diff
@@ -16,97 +19,137 @@ namespace po = boost::program_options;
 
 // available discord channels
 // channels are configured in lib/discord.cfg
-// new channels must be added in that file, here, in discord.h, and in po::options_descriptions below
+// new channels must be added in that file, here, in discord.h, and in
+// po::options_descriptions below
 sstring Discord::CHANNEL_DEATHS;
 sstring Discord::CHANNEL_SYS;
 sstring Discord::CHANNEL_ACHIEVEMENT;
 
 // threshold level for discord mob kill notifications
 int Discord::ACHIEVEMENT_THRESHOLD;
 
+// for thread management
+bool Discord::stop_thread;
+std::thread Discord::messenger_thread;
+std::queue<std::pair<sstring, sstring>> Discord::message_queue;
+std::mutex Discord::queue_mutex;
+
 // read the configuration
 bool Discord::doConfig() {
-    using std::string;
-    string configFile = "discord.cfg";
-
-    std::string empty_string = "";
-
-    po::options_description configOnly("Configuration File Only");
-    configOnly.add_options()
-        ("deaths_webhook",po::value<string>(&CHANNEL_DEATHS)->default_value(empty_string),"see discord.h")
-        ("sys_webhook",po::value<string>(&CHANNEL_SYS)->default_value(empty_string),"see discord.h")
-        ("achieve_webhook",po::value<string>(&CHANNEL_ACHIEVEMENT)->default_value(empty_string),"see discord.h")
-        ("achieve_threshold",po::value<int>(&ACHIEVEMENT_THRESHOLD)->default_value(80),"see discord.h");
-
-    po::options_description config_options;
-    config_options.add(configOnly);
-
-    po::variables_map vm;
-    po::notify(vm);
-    std::ifstream ifs(configFile.c_str());
-
-    if (!ifs.is_open()) {
-        vlogf(LOG_MISC, format("Discord webhooks: Failed to open config file '%s'") % configFile);
-        vlogf(LOG_MISC, format("Discord webhooks: ensure config is located in: '%s'") % std::filesystem::current_path());
-        return false;
-    }
-
-    po::store(parse_config_file(ifs, config_options), vm);
-    po::notify(vm);
-
-    return true;
+  using std::string;
+  // worker thread setup
+  stop_thread = false;
+  messenger_thread = std::thread(Discord::messenger);
+
+  string configFile = "discord.cfg";
+
+  std::string empty_string = "";
+
+  po::options_description configOnly("Configuration File Only");
+  configOnly.add_options()("deaths_webhook",
+    po::value<string>(&CHANNEL_DEATHS)->default_value(empty_string),
+    "see discord.h")("sys_webhook",
+    po::value<string>(&CHANNEL_SYS)->default_value(empty_string),
+    "see discord.h")("achieve_webhook",
+    po::value<string>(&CHANNEL_ACHIEVEMENT)->default_value(empty_string),
+    "see discord.h")("achieve_threshold",
+    po::value<int>(&ACHIEVEMENT_THRESHOLD)->default_value(80), "see discord.h");
```

Comment:
I strongly prefer the way the code was formatted before. This one is breaking lines at totally random places, apparently.

Reply:
Agreed.  Something in my workspace went a bit overboard with the formatting.

Reply:
This is just clang-format not being set up to handle chained function calls like this, I think. I dealt with it in my branch by turning clang-format off and back on via comments, then manually formatting the code in between like this:
```c++
// clang-format off
configOnly.add_options()
  ("deaths_webhook", po::value<string>(&CHANNEL_DEATHS)->default_value(empty_string), "see discord.h") 
  ("sys_webhook", po::value<string>(&CHANNEL_SYS)->default_value(empty_string), "see discord.h") 
  ("achieve_webhook", po::value<string>(&CHANNEL_ACHIEVEMENT)->default_value(empty_string), "see discord.h") 
  ("achieve_threshold", po::value<int>(&ACHIEVEMENT_THRESHOLD)->default_value(80), "see discord.h");
// clang-format on
```

Reply:
neat!